coderd/notifications/dispatch/smtp_util_test.go

```diff
@@ -147,6 +148,10 @@ func (s *Session) Data(r io.Reader) error {
 		return err
 	}
 
+	if s.backend.cfg.FailOnDataFn != nil {
+		return s.backend.cfg.FailOnDataFn()
+	}
```

Comment:
At some point in the future, would it make sense to test with something like `mocktools/go-smtp-mock` instead?

Reply:
I evaluated it, but it does not support authentication.